onnxruntime/test/providers/cpu/llm/attention_op_test.cc

```diff
@@ -0,0 +1,893 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
Why do we only keep a subset of Attention op tests from `contrib_ops/attention_op_test.cc`? There are more than half of them not covered.

Reply:
The test I implemented here are coming from the backend test not the contrib ops. I implemented a couple of cases in 4D to cover all the cases. Then I ran the backend tests in python to make sure it was working on all the cases. CI is running those tests as well.